apps/wallet/src/ui/app/components/accounts/UnlockAccountContext.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { createContext, useContext, useState, type ReactNode, useCallback } from 'react';
+
+import { toast } from 'react-hot-toast';
+import { UnlockAccountModal } from './UnlockAccountModal';
+import { useAccounts } from '../../hooks/useAccounts';
+import { useBackgroundClient } from '../../hooks/useBackgroundClient';
+import { useUnlockMutation } from '../../hooks/useUnlockMutation';
+
+interface UnlockAccountContextType {
+	isUnlockModalOpen: boolean;
+	accountIdToUnlock: string | null;
+	unlockAccount: (accountId: string) => void;
+	isLoading: boolean;
+	lockAccount: (accountId: string) => void;
+	hideUnlockModal: () => void;
+}
+
+const UnlockAccountContext = createContext<UnlockAccountContextType | undefined>(undefined);
+
+export const UnlockAccountProvider: React.FC<{ children: ReactNode }> = ({ children }) => {
```

Comment:
super: It doesn't matter to me how we define/type components, but it would be nice to be consistent. For example, this versus the other provider further above:

`export const OnboardingFormProvider = ({ children }: { children: ReactNode }) =>`

Reply:
not even sure where that came from!